### DISCUSSION TITLE
Solving coupled ODEs with additional exogenous inputs and variable initial conditions

### DISCUSSION THREAD

**Discussion by Rahul-Patel25** [lululxvi/deepxde#1225] [Q&A] [answered]
Hello, 
Thanks for this very useful library.
I am new to DeepXDE and had some doubts about incorporating additional inputs to NN apart from space and time.

I want to solve that system of coupled ODEs t∈[0,10]
dx/dt= -0.5x+y
dy/dt= -x-0.5*y
With Initial condtions x(0)=5, y(0)=9

I did this as below and it works. 
```

import deepxde as dde
import numpy as np

def ode_system(x, y):
    y1, y2 = y[:, 0:1], y[:, 1:]
    dy1_x = dde.grad.jacobian(y, x, i=0)
    dy2_x = dde.grad.jacobian(y, x, i=1)
    return [dy1_x + 0.5*y1 - y2, dy2_x + y1 + 0.5*y2]

def boundary(_, on_initial):
    return on_initial

geom = dde.geometry.TimeDomain(0, 10)
ic1 = dde.icbc.IC(geom, lambda x: 5, boundary, component=0)
ic2 = dde.icbc.IC(geom, lambda x: 9, boundary, component=1)
data = dde.data.PDE(geom, ode_system, [ic1, ic2], 350, 20)

layer_size = [1] + [50] * 3 + [2]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001)
losshistory, train_state = model.train(iterations=20000)
dde.saveplot(losshistory, train_state, issave=True, isplot=True)
```

**Question 1.**
Now, I want **0.5** in second equation to be a variable (lets say **Q** and it doesn't change with time), whose value will be an input to the NN. i.e NN will predict the outputs (x,y) given the inputs (time,**Q**).  Ideally I would train the NN with time t∈[0,10] and **Q**∈[0,1], i.e with Q as an additional exogenous input to the NN. How can I achieve this using DeepXDE. 
dx/dt= -0.5x+y
dy/dt= -x-**Q***y
where t∈[0,10] and Q∈[0,1]

**Question 2.**
How can I model the same for initial conditions that are not fixed beforehand. I would like to give the initial conditions (**x0**,**y0**) as an input to the NN.

dx/dt= -0.5x+y
dy/dt= -x-**Q***y
With Initial condtions x(0)=**x0**, y(0)=**y0**

The NN should predict outputs (x,y) given the inputs(time,**Q**,**x0**,**y0**). I want to train the NN for the inputs time t∈[0,10], **Q**∈[0,1], **x0**∈[2,8],**y0**∈[3,7]

Let me know how I should proceed.

**Comment by Rahul-Patel25** [comment, marked as answer]
I was able to achieve the objective. I considered Hypercube as the geometry and added the additional inputs as the other dimensions of the hypercube and changing the boundary conditions suitably.

**Reply by John-Silva-29** [reply]
can you share your code, i am having the same problem. I want to use the PINN model in a process control framework, therefore i need a model that is able to simulate varying boundary conditions and parameter values
